src/Findologic/Api/FindologicSearchService.php

```diff
@@ -124,7 +126,7 @@ protected function handleRequest(
         $this->sortingService->handleRequest($event, $requestHandler);
     }
 
-    protected function allowRequest(ProductListingCriteriaEvent $event): bool
+    public function allowRequest(ProductListingCriteriaEvent $event): bool
```

Comment:
It may be convenient to have this public, but I am against that. The `FindologicSearchService` should be aware of its state and must know itself if it should trigger or not. Make sure this method is called in `doFilter`. Every public method of this class should be usable, without having to think about any state.

In `SearchController::filter` you currently use this method to check if Findologic should be used or not. This shouldn't be strictly tied to this class. Instead find a similar approach like in `ProductListingRoute::load`:
```php
public function __construct()
{
    // ...
    $this->config = $config ?? new Config($systemConfigService, $serviceConfigResource);
}

public function load(
    string $categoryId,
    Request $request,
    SalesChannelContext $salesChannelContext
): ProductListingRouteResponse {
    $this->config->initializeBySalesChannel($salesChannelContext->getSalesChannel()->getId());
    $shouldHandleRequest = Utils::shouldHandleRequest(
        $request,
        $salesChannelContext->getContext(),
        $this->serviceConfigResource,
        $this->config,
        true
    );
    // ...
}
```